src/mongo/delegates/Updates.class.php

```diff
@@ -1204,6 +1233,14 @@ public function setTransactionLog(TransactionLog $transactionLog)
         $this->transactionLog = $transactionLog;
     }
 
+    /**
+     * Register save changes event hooks
+     * @param IEventHook $hook
+     */
+    public function registerSaveChangesEventHook(IEventHook $hook)
+    {
+        $this->saveChangesHooks[] = $hook;
+    }
```

Comment:
We are designing this with the intention of adding more hooks though - Driver:: registerHook() has been written with other hook types in mind.  It seems strange to have parts of the code prepared for new hook types, but other parts baked in like this.  If other hooks are on the roadmap then I would favour doing this earlier, rather than having to take it out later when we want to add another hook.

If we are sure we are going to be adding more hooks, and those hooks could be across multiple delegates, then maybe we should be breaking out into its own class which can be instantiated and shared between delegates.